### PR TITLE
Support receiving multiple DownloadParams in DownloadService

### PR DESCRIPTION
**Pull request by barbelity**

### CODE REVIEW

File: artifactory/services/download.go [hunk 1]
```diff
@@ -133,7 +143,7 @@ func produceTasks(items []utils.ResultItem, downloadParams DownloadParams, produ
 		}
 		if v.Type != "folder" {
 			// Add a task, task is a function of type TaskFunc which later on will be executed by other go routine, the communication is done using channels.
```
Comment (eyalbe4) [inline artifactory/services/download.go]: "// Add a task, task is a..." --> "// Add a task. A task is a..."

File: artifactory/manager.go [hunk 2]
```diff
@@ -83,14 +83,14 @@ func (sm *ArtifactoryServicesManager) ReadRemoteFile(readPath string) (io.ReadCl
 	return readFileService.ReadRemoteFile(readPath)
 }
 
-func (sm *ArtifactoryServicesManager) DownloadFiles(params services.DownloadParams) ([]utils.FileInfo, int, error) {
+func (sm *ArtifactoryServicesManager) DownloadFiles(params ...services.DownloadParams) ([]utils.FileInfo, int, error) {
```
Comment (eyalbe4) [inline artifactory/manager.go]: The README should be updated with the new API signature.

File: artifactory/services/download.go [hunk 3]
```diff
@@ -133,7 +143,7 @@ func produceTasks(items []utils.ResultItem, downloadParams DownloadParams, produ
...
-			// The second argument is a error handling func in case the taskFunc return an error.
+			// The second argument is an error handling func in case the taskFunc return an error.
 			tasksCount++
 			producer.AddTaskWithError(fileHandler(tempData), errorsQueue.AddError)
 			// We don't want to create directories which are created explicitly by download files when the --include-dirs flag is used.
```
Comment (eyalbe4) [inline artifactory/services/download.go]: "...when the --include-dirs flag is used." --> "...when **ArtifactoryCommonParams.IncludeDirs** is used."

Comment (barbelity) [reply artifactory/services/download.go]: Changed.

File: artifactory/services/download.go (hunk 1 above)
Comment (barbelity) [reply artifactory/services/download.go]: Changed.

File: artifactory/manager.go (hunk 2 above)
Comment (barbelity) [reply artifactory/manager.go]: The README is not affected by this change.

**Pull request merged**